tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Tests\Util;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bundle\MakerBundle\Util\PhpServicesCreator;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Loader\PhpFileLoader;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\Finder\Finder;
+
+// For testing parameters conversion
+\define('GLOBAL_CONSTANT', 'value');
+
+class PhpServicesCreatorTest extends TestCase
+{
+    // For testing parameters conversion
+    const CONSTANT_NAME = 1;
+
+    /**
+     * @dataProvider getConfigurationFiles
+     */
+    public function testConvertionYamlServices(string $yamlFile, string $phpFile, string $nameTestCase)
+    {
+        $creator = new PhpServicesCreator();
+        $this->assertSame($phpFile, $creator->convert($yamlFile));
+        /*
+         * test if default comments and configuration are written but ressources can't be loaded.
+         * the container building is tested by the testLoadResources() above.
+         */
+        if ('services_load_ressources' !== $nameTestCase) {
+            $yamlContainerBuilder = new ContainerBuilder();
+            $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source'));
+            $yamlLoader->load($nameTestCase.'.yaml');
+
+            $phpContainerBuilder = new ContainerBuilder();
+            $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/'));
+            $phpLoader->load($nameTestCase.'.php');
```

Comment:
For loading the files, we only have the file *contents*, not the original filenames. You've worked around that by passing `$nameTestCase`, which is kind of a hack - a test shouldn't care about what its test case name is. Here's what I would do. For both YAML and PHP:

1) Create a temporary file - like `$tmpFilename = tempnam(sys_get_temp_dir(), 'maker_');`
2) Read that file:

```php
// it's legal to pass nothing to FileLocator and then pass it an absolute path file
$yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator());
$yamlLoader->load($tmpFilename);
```

3) Delete the file -`unlink($tmpFilename)`

Reply:
Done.